### PR TITLE
Add support for runtime notebook hosted URL

### PR DESCRIPTION
**Pull request by xaka**

### CODE REVIEW

File: src/scripts/embed.js [hunk 1]
```diff
@@ -165,7 +167,7 @@ Notebook.unsubscribe = function (fn) {
  */
 Notebook.prototype._makeFrame = function (el, options) {
   var notebook = this;
-  var src      = NOTEBOOK_URL + '/embed.html';
+  var src      = (options.url || NOTEBOOK_URL) + '/embed.html';
```
Comment (blakeembrey) [inline src/scripts/embed.js]: Should this be the complete path to the `embed.html` or just the parent directory like it is now?

File: src/scripts/embed.js [hunk 2]
```diff
@@ -401,8 +403,20 @@ Notebook.prototype.trigger = function (name /*, ..args */) {
     return this;
   }
 
+  // For relative URLs, such as /proxy/notebook, we need
+  // to re-construct absolute URLs as browsers would do for us,
+  // except for Window.postMessage, which expects origin, not URL.
+
+  var windowUrl   = url.parse(window.location.href);
+  var notebookUrl = url.parse(this.options.url || NOTEBOOK_URL);
```
Comment (blakeembrey) [inline src/scripts/embed.js]: I think `this.options` should be `this.options = extend({ url: NOTEBOOK_URL + '/embed.html' }, options)` in the constructor. That way, we just use that object properly and just parse `this.options.url`.

File: src/scripts/embed.js (hunk 1 above)
Comment (xaka) [inline src/scripts/embed.js]: The parent directory as with `NOTEBOOK_URL`. It makes sense as there are more than just one file to serve.

File: src/scripts/embed.js (hunk 2 above)
Comment (xaka) [inline src/scripts/embed.js]: Alright. I've moved `NOTEBOOK_URL` under `defaultOptions.url` so it's much cleaner now.

File: src/scripts/embed.js [hunk 3]
```diff
@@ -61,6 +59,7 @@ var getDataAttributes = function (el) {
  * @type {Object}
  */
 var defaultOptions = {
+  url:     process.env.application.url, // Location to load the notebook from
   id:      null, // Initial id to pull content from
   content: '',   // Fallback content in case of no id
   style:   {},   // Set styles on the iframe
```
Comment (xaka) [inline src/scripts/embed.js]: I'd align comment lines, but `jshint` complains line is too long :)

File: src/scripts/embed.js [hunk 4]
```diff
@@ -401,8 +400,20 @@ Notebook.prototype.trigger = function (name /*, ..args */) {
...
+  // to re-construct absolute URLs as browsers would do for us,
+  // except for Window.postMessage, which expects origin, not URL.
+
+  var windowUrl   = url.parse(window.location.href);
+  var notebookUrl = url.parse(this.options.url);
+  var origin      = [
+    (notebookUrl.protocol || windowUrl.protocol),
+    '//',
+    (notebookUrl.host || windowUrl.host)
+  ].join('');
```
Comment (blakeembrey) [inline src/scripts/embed.js]: So this part of the code, I assume it's to handle partially completed URLs? Could we just set it at the beginning to be complete then and make this line simpler? For example, `options.url = url.resolve(window.location.href, options.url)`?

Comment (xaka) [inline src/scripts/embed.js]: Switched to `url.resolve`. Works like a charm.

**Comment by blakeembrey** [comment]
LGTM. I'm going to merge and fix it so we're not mutating the `options` object, but it looks good.

**Pull request closed without merging**